tests/test_dataset.py

```diff
@@ -29,11 +29,19 @@
 HYPOTHESIS_MAX_EXAMPLES = 20
 
 
-@pytest.fixture()
+@pytest.fixture
 def grid():
     return pyvista.UnstructuredGrid(examples.hexbeamfile)
 
 
+@pytest.fixture
+def clean_up_type_overrides():
+    """Reset type overrides on teardown."""
+    yield
```

Comment:
Do we need this line?

Reply:
Yes, otherwise the cleanup would run _before_ the test. Turning the fixture into a generating function splits it into "setup - yield - teardown".